google/utils.go

```diff
@@ -28,14 +28,32 @@ func getRegionFromZone(zone string) string {
 
 // getRegion reads the "region" field from the given resource data and falls
 // back to the provider's value if not given. If the provider's value is not
-// given, an error is returned.
+// given, checks for "zone" in either the given resource data or provider,
+// and extracts region from zone.  If "zone" is not provided, returns an
+// error.
 func getRegion(d *schema.ResourceData, config *Config) (string, error) {
```

Comment:
Probably worth it to add a unit test for getRegion and getZone.

Reply:
I'm sure you're right - I'll get on that.  :)

Reply:
Done.